packages/@sanity/code-input/src/CodeInput.js

```diff
@@ -84,7 +127,11 @@ export default class Code extends React.PureComponent {
         onChange={this.handleCodeChange}
         name={`${this._inputId}__aceEditor`}
         value={value.code || ''}
-        editorProps={{$blockScrolling: true}}
+        markers={value.highlightedLines ? this.createMarkers(value.highlightedLines) : null}
```

Comment:
Should also check if the array is empty here? Seems one of the markers are erroneously rendered when you toggle off the last highlighted line.